### PR TITLE
fix: AngledGate equality checks angles

### PR DESCRIPTION
**Pull request by speller26**
*Issue #, if available:*
#213

*Description of changes:*

Currently `AngledGate` does not check for angle equality (just type equality) in `__eq__`. This change fixes this.

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-sdk-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by licedric** [comment]
Can you briefly describe this change, e.g. what was the previous behavior?

File: src/braket/circuits/angled_gate.py [hunk 1]
```diff
@@ -52,5 +53,10 @@ def angle(self) -> float:
         """
         return self._angle
 
+    def __eq__(self, other):
+        if isinstance(other, AngledGate):
+            return self.name == other.name and math.isclose(self.angle, other.angle)
```
Comment (licedric) [inline src/braket/circuits/angled_gate.py:58]: I was wondering if angles differing by 4pi should be considered equal as well, but that's probably a story for another day (and it complicates the stuff you can do here).

Comment (licedric) [reply src/braket/circuits/angled_gate.py:58]: Also, are we sure we want to treat close angles as equal? This does imply that equality is not transitive.

Comment (speller26) [reply src/braket/circuits/angled_gate.py:58]: I'm just paranoid about floating points; I do the same for matrices in #217 . Besides, the default tolerance of 1e09 should be more than enough.

**Review by licedric — APPROVED** [review]

**Pull request merged**